core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java

```diff
@@ -608,6 +608,10 @@ public MetaResultSet getTableTypes(ConnectionHandle ch) {
             new ArrayList<List<Object>>());
     boolean done = fetchMaxRowCount == 0 || rows.size() < fetchMaxRowCount;
     @SuppressWarnings("unchecked") List<Object> rows1 = (List<Object>) rows;
+    if (done) {
```

Comment:
The logic looks still vulnerable. If the first iterator is not "done", then we execute second query the result will be wrong. I think the correct fix should relate to the CLOSE phase of ResultSet instances in client side, but I tested against calling `resultSet.close()` before executing the second query and nothing happened. Maybe something was wrong there.

Reply:
@zhztheplayer Thanks for you quick look again.  And i don't know what you means

> calling resultSet.close() before executing the second query and nothing happened

Since I think there is no need to call ResultSet#close.

And I add an unit test for fetching more than 100 result. Please take a look. Thanks.

Reply:
It seems that `stmt.setResultSet(null)` should be called by the operation of closing the ResultSet object on client side, because inside the method `CalciteMetaImpl#fetch` we can barely know exactly when we should execute a new query (please correct me if I am wrong). 

For example, you can check following test case:
 ```java
  @Test public void testRemoteExecuteMultiQueriesWithMoreResultInSameStatement() throws Exception {
    try (Connection remoteConnection = getRemoteConnection()) {
      final Statement statement = remoteConnection.createStatement();
      StringBuffer stringBuffer = new StringBuffer();
      stringBuffer.append("values ");
      final int resultSize = 1000;
      for (int i = 0; i < resultSize; i++) {
        stringBuffer.append("(").append(i).append(", ").append(i + 1).append(")");
        if (i != resultSize - 1) {
          stringBuffer.append(", ");
        }
      }
      final String sql = stringBuffer.toString();
      ResultSet resultSet = statement.executeQuery(sql);

      resultSet = statement.executeQuery("values (1, 'a')");
      int n = 0;
      while (resultSet.next()) {
        ++n;
      }
      assertThat(n, equalTo(1));
    }
  }
```